core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -108,31 +99,26 @@ public static void init(String sitePropertiesFile) {
     defaultProps.setProperty(PropertyKey.USER_NETWORK_NETTY_CHANNEL.toString(),
         String.valueOf(ChannelType.defaultType()));
 
-    String confPaths;
-    // If site conf is overwritten in system properties, overwrite the default setting
-    if (System.getProperty(PropertyKey.SITE_CONF_DIR.toString()) != null) {
-      confPaths = System.getProperty(PropertyKey.SITE_CONF_DIR.toString());
-    } else {
-      confPaths = defaultProps.getProperty(PropertyKey.SITE_CONF_DIR.toString());
-    }
-    String[] confPathList = confPaths.split(",");
-
-    // Load site specific properties file
-    Properties siteProps = ConfigurationUtils
-        .searchPropertiesFile(sitePropertiesFile, confPathList);
-
     // Load system properties
     Properties systemProps = new Properties();
     systemProps.putAll(System.getProperties());
 
     // Now lets combine, order matters here
     PROPERTIES.clear();
     merge(defaultProps);
-    if (siteProps != null) {
-      merge(siteProps);
-    }
     merge(systemProps);
 
+    // Load site specific properties file if not in test mode
+    if (!getBoolean(PropertyKey.TEST_MODE)) {
+      String confPaths = get(PropertyKey.SITE_CONF_DIR);
+      String[] confPathList = confPaths.split(",");
+      Properties siteProps = ConfigurationUtils.searchPropertiesFile(SITE_PROPERTIES, confPathList);
+      if (siteProps != null) {
+        merge(siteProps);
+        merge(systemProps);
```

Comment:
should we put this block of new code between `merge(defaultProps)` and `merge(systemProps)` so that we don't need to call `merge(systemProps)` a second time?